Validation extension point (#275)
projectcaluma/caluma: README.md

Arguments:
* `mutation`: mutation class
* `info`: resolver info, whereas `info.context` is the [http request](https://docs.djangoproject.com/en/1.11/ref/request-response/#httprequest-objects) and user can be accessed through `info.context.user`
* `instance`: instance being edited by specific mutation

Save your permission module as `permissions.py` and inject it as Docker volume to path `/app/caluma/extensions/permissions.py`,
see [docker-compose.yml](https://github.com/projectcaluma/caluma/blob/master/docker-compose.yml) for an example.

Afterwards you can configure it in `PERMISSION_CLASSES` as `caluma.extensions.permissions.CustomPermission`.

##### Validation classes

Validation classes validate or amend input data of any mutation. It completes permission classes which allow execution of mutations based on updating object whereas validation classes

> process input data.

Reviewer: Validation classes can validate or amend input data of any mutation. Each mutation is processed in two steps:

1. Permission classes check if a given mutation is allowed based on the object being mutated, and
2. Validation classes process (and potentially amend) input data of a given mutation

> A custom validation class defining validation per mutation can be defined like the following:

Reviewer: ```suggestion
A custom validation class defining validations for various mutations looks like this:
```

```python
from caluma.core.validations import BaseValidation, validation_for
from caluma.form.schema import SaveForm
from caluma.mutation import Mutation
from rest_framework import exceptions

class CustomValidation(BaseValidation):
    @validation_for(Mutation)
    def validate_mutation(self, mutation, data, info):
        # add your default specific validation code here
        return data

    @validation_for(SaveForm)
    def validate_save_form(self, mutation, data, info):
        if data['meta'] and info.context.group != 'admin':
          raise exceptions.ValidationException('May not change meta on form')
        return data
```

Arguments:
* `mutation`: mutation class

> * `data`: input data whereas relation are resolved (e.g. a form ID is represented as actual form object)

Reviewer: ```suggestion
* `data`: input data with resolved relationships (e.g. a form ID is represented as actual form object)
```
